packages/wrangler/src/ai/fetcher.ts

```diff
@@ -13,24 +14,30 @@ export default function (env) {
 }
 `;
 
-export async function AIFetcher(request: Request): Promise<Response> {
-	const accountId = await getAccountId();
-
-	const reqHeaders = new Headers(request.headers);
-	reqHeaders.delete("Host");
-	reqHeaders.delete("Content-Length");
-	reqHeaders.set("X-Forwarded", request.url);
-
-	const res = await performApiFetch(`/accounts/${accountId}/ai/run/proxy`, {
-		method: request.method,
-		headers: Object.fromEntries(reqHeaders.entries()),
-		body: request.body,
-		duplex: "half",
-	});
-
-	const respHeaders = new Headers(res.headers);
-	respHeaders.delete("Host");
-	respHeaders.delete("Content-Length");
-
-	return new Response(res.body, { status: res.status, headers: respHeaders });
+export function getAIFetcher(complianceConfig: ComplianceConfig) {
```

Comment:
This fetcher is run in a context that doesn't have access to the config.
But the config is available at the time we import this function, so by converting this to a factory, we can capture the config in the context of the fetcher.